File: matlab_examples/CalibrationResultsUpdated.m
function CalibrationResultsUpdated(src, event)
    subplot(2, 1, 1);   
    
    if (isempty(event.TimeAxis) == 0)
        x = double(event.TimeAxis);
        plot(x, double(event.TimeBlock));  
        xlim([x(1) x(length(x))]);
    end
               
    title('Time'); 
    legend(sprintf('Time elapsed (s): %.2f', event.CurrentTimeElapsed));
    
    subplot(2, 1, 2);
    
    if (isempty(event.FrequencyAxis) == 0)        
        plot(double(event.FrequencyAxis), double(event.ResponseAutoSpectrum));    
        xlim([500 1500])  ;
    end
   
    title('Averaged Autospectrum (RMS)');    
    legend(sprintf('Average: %d/%d', event.CurrentAverageNumber,event.TotalAverages));
     
    drawnow;
end

File: python_examples/CalibrationResultsUpdated.py


def calibrationresultsupdated(source, args):
    if args.CurrentAverageNumber < args.TotalAverages:
        print('\rCurrent time elapsed: {0:.2f} s  -  Current average: {1} / {2}'.format(args.CurrentTimeElapsed, args.CurrentAverageNumber, args.TotalAverages), end='')
    else:
        print('')





